### ISSUE TITLE
Need Help for iOS Simulator and MacOS Build of OpenSSL

### ISSUE THREAD

**Issue by serajahmad01** [openssl/openssl#18921]
I am working on a project where I have to build OpenSSL "3.0.5" version for multiple targets related to MacOS and iOS.

I am able to successfully build the OpenSSL "1.1.1x" version by using targets mentioned in "20-ios-tvos-cross.conf".
I used to give commands like the below:
```
./Configure macosx-cross-x86_64 --prefix=gen/out/bin/Debug-MacOSX-x86_64 no-shared no-async -fembed-bitcode-marker enable-ec_nistp_64_gcc_128
```

Now for version "3.0.5" when I give the above command, I am getting the below error:

```
./Configure macosx-cross-x86_64 --prefix=gen/out/bin/Debug-MacOSX-x86_64 no-shared no-async -fembed-bitcode-marker enable-ec_nistp_64_gcc_128
Undefined subroutine &main::asm called at (eval 15) line 46.
Configuring OpenSSL version 3.0.5 for target macosx-cross-x86_64
Using os-specific seed configuration
Usage: Configure [no-<cipher> ...] [enable-<cipher> ...] [-Dxxx] [-lxxx] [-Lxxx] [-fxxx] [-Kxxx] [no-hw-xxx|no-hw] [[no-]threads] [[no-]shared] [[no-]zlib|zlib-dynamic] [no-asm] [no-egd] [sctp] [386] [--prefix=DIR] [--openssldir=OPENSSLDIR] [--with-xxx[=vvv]] [--config=FILE] os/compiler[:flags]

pick os/compiler from:
...
android-armeabi android-mips android-mips64 android-x86 android-x86_64
android64 android64-aarch64 android64-mips64 android64-x86_64 bsdi-elf-gcc cc
darwin-i386 darwin-i386-cc darwin-ppc darwin-ppc-cc darwin64-arm64
darwin64-arm64-cc darwin64-debug-test-64-clang darwin64-ppc darwin64-ppc-cc
darwin64-x86_64 darwin64-x86_64-cc gcc haiku-x86 haiku-x86_64 hpux-ia64-cc
...
```

When I give the "os/compiler" details in the above command, I get the below error:

```
./Configure macosx-cross-x86_64 --prefix=gen/out/bin/Debug-MacOSX-x86_64 no-shared no-async -fembed-bitcode-marker enable-ec_nistp_64_gcc_128 darwin64-x86_64-cc
Undefined subroutine &main::asm called at (eval 15) line 46.

Failure!  build file wasn't produced.
Please read INSTALL.md and associated NOTES-* files.  You may also have to
look over your available compiler tool chain or change your configuration.

target already defined - macosx-cross-x86_64 (offending arg: darwin64-x86_64-cc)
```

Now, my question is I want to build the OpenSSL using the targets defined in my "20-ios-tvos-cross.conf" and without "OS/Compiler" flags? I have tried to find the solution over the internet but no luck.
Will really appreciate the help in this. Feel free to ask any questions you have.

Thanks,
Seraj

**Comment by tom-cosgrove-arm** [comment]
`20-ios-tvos-cross.conf` appears to come from a different project, [OpenSSL-for-iPhone](https://github.com/x2on/OpenSSL-for-iPhone/blob/master/config/20-ios-tvos-cross.conf) - have you tried asking there?

The equivalent file in OpenSSL is [15-ios.conf](https://github.com/openssl/openssl/blob/master/Configurations/15-ios.conf)

**Comment by serajahmad01** [comment]
Thanks @tom-cosgrove-arm for the reply on this.
My question is how can I add more targets specifically related to MacOS as you see in this "[OpenSSL-for-iPhone](https://github.com/x2on/OpenSSL-for-iPhone/blob/master/config/20-ios-tvos-cross.conf) " file.

Thanks,
Seraj

**Comment by VityaSchel** [comment]
> Thanks @tom-cosgrove-arm for the reply on this. My question is how can I add more targets specifically related to MacOS as you see in this "[OpenSSL-for-iPhone](https://github.com/x2on/OpenSSL-for-iPhone/blob/master/config/20-ios-tvos-cross.conf) " file.
> 
> Thanks, Seraj

Hi, have you found a solution?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
